src/inmanta/app.py

```diff
@@ -591,40 +601,180 @@ def export(options: argparse.Namespace) -> None:
+            metadata=metadata,
+            model_export=options.model_export,
+            export_plugin=options.export_plugin,
+            partial_compile=options.partial_compile,
+            resource_sets_to_remove=options.delete_resource_set,
+        )
 
-    if options.deploy:
+    if not summary_reporter.is_failure() and options.deploy:
+        version = results[0]
         conn = protocol.SyncClient("compiler")
         LOGGER.info("Triggering deploy for version %d" % version)
         tid = cfg_env.get()
         agent_trigger_method = const.AgentTriggerMethod.get_agent_trigger_method(options.full_deploy)
         conn.release_version(tid, version, True, agent_trigger_method)
 
+    summary_reporter.print_summary_and_exit(show_stack_traces=options.errors)
+
+
+class Color(enum.Enum):
+    RED = "red"
+    GREEN = "green"
+
+
+@dataclasses.dataclass
+class ExceptionCollector:
+    """
+    This class defines a context manager that captures any unhandled exception raised within the context.
+    """
+
+    exception: Optional[Exception] = None
+
+    def has_exception(self) -> bool:
+        return self.exception is not None
+
+    @contextlib.contextmanager
+    def capture(self) -> abc.Iterator[None]:
+        """
+        Record any exceptions raised within the context manager. The exception will not be re-raised.
+        """
+        try:
+            yield self
+        except Exception as e:
+            self.exception = e
+
+
+class CompileSummaryReporter:
+    """
+    Contains the logic to print a summary at the end of the `inmanta compile` of `inmanta export`
+    command that provides an overview on whether the command was successful or not.
+    """
+
+    def __init__(self) -> None:
+        self.compiler_exception: ExceptionCollector = ExceptionCollector()
+        self.exporter_exception: ExceptionCollector = ExceptionCollector()
+
+    def is_failure(self) -> bool:
+        """
+        Return iff an exception has occurred during the compile or export stage.
+        """
+        return self.compiler_exception.has_exception() or self.exporter_exception.has_exception()
+
+    def _get_global_status(self) -> str:
+        """
+        Return the global status of the run.
+        """
+        if self.compiler_exception.has_exception():
+            return "COMPILATION FAILURE"
+        elif self.exporter_exception.has_exception():
+            return "EXPORT FAILURE"
+        else:
+            return "SUCCESS"
+
+    def _get_header(self, header_text: str) -> str:
+        """
+        Return a header for the summary with the given header_text.
+        """
+        terminal_width = shutil.get_terminal_size()[0]
+        minimal_header = f"= {header_text.upper()} ="
+        length_minimal_header = len(minimal_header)
+        if terminal_width <= length_minimal_header:
+            return minimal_header
+        else:
+            nr_equals_signs_to_add_each_side = int((terminal_width - length_minimal_header) / 2)
```

Comment:
I don't think that's an issue. We have to choose here between a header that is either one character shorter than the total terminal width or having an asymmetrical header. I chose for the first option, because I find an asymmetrical header quite ugly.